hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/MongoBackendResolver.java

```diff
@@ -16,39 +18,31 @@
 import org.slf4j.LoggerFactory;
 
 public class MongoBackendResolver extends AbstractBackendResolver {
+
   private static final Logger LOGGER = LoggerFactory.getLogger(MongoBackendResolver.class);
   private static final String RAW_MONGO_NAMESPACE = "NAMESPACE";
-  private static final String MONGO_ADDRESS_ATTR = RawSpanConstants.getValue(Mongo.MONGO_ADDRESS);
-  private static final String MONGO_URL_ATTR = RawSpanConstants.getValue(Mongo.MONGO_URL);
 
   public MongoBackendResolver(FQNResolver fqnResolver) {
     super(fqnResolver);
   }
 
   @Override
   public Optional<Entity> resolveEntity(Event event, StructuredTraceGraph structuredTraceGraph) {
-    if (SpanAttributeUtils.containsAttributeKey(event, MONGO_ADDRESS_ATTR) ||
-        SpanAttributeUtils.containsAttributeKey(event, MONGO_URL_ATTR)) {
-      String backendURI = null;
-      // check if this is from uninstrumeted agent first
-      if (SpanAttributeUtils.containsAttributeKey(event, MONGO_ADDRESS_ATTR)) {
-        backendURI = SpanAttributeUtils.getStringAttribute(event, MONGO_ADDRESS_ATTR);
-      } else if (SpanAttributeUtils.containsAttributeKey(event, MONGO_URL_ATTR)) {
-        backendURI = SpanAttributeUtils.getStringAttribute(event, MONGO_URL_ATTR);
-      }
-
-      if (backendURI == null) {
-        LOGGER.debug("Detected MONGO span event, but unable to extract the URI. Span Event: {}",
-            event);
-        return Optional.empty();
-      }
-
-      final Builder entityBuilder = getBackendEntityBuilder(BackendType.MONGO, backendURI, event);
-      setAttributeIfExist(event, entityBuilder, RAW_MONGO_NAMESPACE);
-      setAttributeIfExist(event, entityBuilder, RawSpanConstants.getValue(Mongo.MONGO_NAMESPACE));
-      setAttributeIfExist(event, entityBuilder, RawSpanConstants.getValue(Mongo.MONGO_OPERATION));
-      return Optional.of(entityBuilder.build());
+    Optional<String> backendURI = DbAttributeUtils.getMongoURI(event);
+
+    if (backendURI.isEmpty()) {
+      return Optional.empty();
     }
-    return Optional.empty();
+
+    if (StringUtils.isEmpty(backendURI.get())) {
+      LOGGER.warn("Unable to infer a redis backend from event: {}", event);
+      return Optional.empty();
+    }
+
+    final Builder entityBuilder = getBackendEntityBuilder(BackendType.MONGO, backendURI.get(), event);
+    setAttributeIfExist(event, entityBuilder, RAW_MONGO_NAMESPACE);
```

Comment:
Q: In which format, do we expect to get this `RAW_MONGO_NAMESPACE`? If, it is custom as well?
Can you move it under getTagsForMongoNamespace in DbAttributesUtils? So, if we have to extend it to another format, it becomes easy?

Reply:
That information is not retained in the code, so I am not sure. 
The reason I didn't move it under DbAttributesUtils #getTagsForMongoNamespace, is that both these constants are being separately set in the existing code. 
```setAttributeIfExist(event, entityBuilder, RAW_MONGO_NAMESPACE);	
      setAttributeIfExist(event, entityBuilder, RawSpanConstants.getValue(Mongo.MONGO_NAMESPACE));```